src/test/java/org/openrewrite/quarkus/ChangeQuarkusPropertyKeyTest.java

```diff
@@ -0,0 +1,289 @@
+                        strategy: test
+              '%dev':
+                quarkus:
+                  hibernate-search-orm:
+                    automatic-indexing:
+                      synchronization:
+                        strategy: test
+                    unitname:
+                      automatic-indexing:
+                        synchronization:
+                          strategy: test
+              '%prod':
+                quarkus:
+                  hibernate-search-orm:
+                    indexing:
+                      plan:
+                        synchronization:
+                          strategy: test
+                    unitname:
+                      indexing:
+                        plan:
+                          synchronization:
+                            strategy: test
+              '%staging':
+                quarkus:
+                  hibernate-search-orm:
+                    automatic-indexing:
+                      synchronization:
+                        strategy: test
+                    unitname:
+                      automatic-indexing:
+                        synchronization:
+                          strategy: test
+              """;
+
+            Recipe recipe = new ChangeQuarkusPropertyKey("quarkus\\.hibernate-search-orm(\\..*)?\\.automatic-indexing\\.synchronization\\.strategy", "quarkus.hibernate-search-orm$1.indexing.plan.synchronization.strategy", null, "prod", false, null);
+
+            rewriteRun(spec -> spec.recipe(recipe), yaml(sourceYaml, after, spec -> spec.path("src/main/resources/application.yaml")));
+        }
+
+        @Test
+        void changeAllProfiles() {
+            @Language("yml")
+            String after = """
+              quarkus:
+                hibernate-search-orm:
+                  unitname:
+                    indexing:
+                      plan:
+                        synchronization:
+                          strategy: test
+                  indexing:
+                    plan:
+                      synchronization:
+                        strategy: test
+              '%dev':
+                quarkus:
+                  hibernate-search-orm:
+                    indexing:
+                      plan:
+                        synchronization:
+                          strategy: test
+                    unitname:
+                      indexing:
+                        plan:
+                          synchronization:
+                            strategy: test
+              '%staging,prod':
+                quarkus:
+                  hibernate-search-orm:
+                    indexing:
+                      plan:
+                        synchronization:
+                          strategy: test
+                    unitname:
+                      indexing:
+                        plan:
+                          synchronization:
+                            strategy: test
+              """;
```

Comment:
When I step through with the debugger, the value passed into `MergeYaml` from `newProperties.toString()` is 
```yaml
quarkus:
  hibernate-search-orm:
    indexing:
      plan:
        synchronization:
          strategy: test
'%dev':
  quarkus:
    hibernate-search-orm:
      automatic-indexing:
        synchronization:
          strategy: test
```
That seems at odds with the values expected here. Could that be the cause of the test failure we're seeing for this one?